backend-api/src/functions/asyncToken/requestService/requestService.ts

```diff
@@ -1,61 +1,80 @@
+import { Result, errorResult, successResult } from "../../utils/result";
+import { APIGatewayProxyEventHeaders } from "aws-lambda";
 import { IDecodedClientSecrets } from "../../services/clientRegistryService/clientRegistryService";
-import { errorResult, Result, successResult } from "../../utils/result";
-import { APIGatewayProxyEvent } from "aws-lambda";
-
-export class RequestService implements IProcessRequest {
-  processRequest = (
-    request: APIGatewayProxyEvent,
-  ): Result<IDecodedClientSecrets> => {
-    const requestBody = request.body;
-    const authorizationHeader = request.headers["Authorization"];
-
-    if (!this.isRequestBodyValid(requestBody)) {
-      return errorResult("Invalid grant_type");
-    }
-
-    if (!authorizationHeader) {
-      return errorResult("Invalid authorization header");
-    }
-
-    if (!authorizationHeader.startsWith("Basic ")) {
-      return errorResult("Invalid authorization header");
-    }
-
-    const base64EncodedCredential = authorizationHeader.split(" ")[1];
-    const base64DecodedCredential = Buffer.from(
-      base64EncodedCredential,
-      "base64",
-    ).toString("utf-8");
-    const [clientId, clientSecret] = base64DecodedCredential.split(":");
-
-    if (!clientId || !clientSecret) {
-      return errorResult("Client secret incorrectly formatted");
-    }
-
-    return successResult({ clientId, clientSecret });
-  };
-
-  private isRequestBodyValid = (body: string | null): boolean => {
-    if (body == null) {
-      return false;
-    }
-
-    const { grant_type } = JSON.parse(body);
-    if (grant_type !== "client_credentials") {
-      return false;
-    }
-
-    return true;
-  };
-}
-
-export interface IProcessRequest {
-  processRequest: (
-    request: APIGatewayProxyEvent,
-  ) => Result<IDecodedClientSecrets>;
-}
 
 export interface IDecodedAuthorizationHeader {
   clientId: string;
   clientSecret: string;
 }
+
+interface IRequestService {
+  getClientCredentials: (
+    headers: APIGatewayProxyEventHeaders,
+  ) => Result<IDecodedAuthorizationHeader>;
+  validateBody: (body: string | null) => Result<null>;
+}
+export const requestService: IRequestService = {
+  getClientCredentials: (headers: APIGatewayProxyEventHeaders) =>
+    getClientCredentials(headers),
+  validateBody: (body: string | null) => validateBody(body),
+};
+
+const getClientCredentials = (
+  headers: APIGatewayProxyEventHeaders,
+): Result<IDecodedClientSecrets> => {
+  const authorizationHeader = headers.Authorization;
+  if (!authorizationHeader) {
+    return errorResult({
```

Comment:
As discussed, please can you check we have test coverage of this

Reply:
done